pkg/machine/qemu/machine.go

```diff
@@ -930,32 +974,42 @@ func (v *MachineVM) stopLocked() error {
 
 // NewQMPMonitor creates the monitor subsection of our vm
 func NewQMPMonitor(network, name string, timeout time.Duration) (Monitor, error) {
-	rtDir, err := getRuntimeDir()
+	if timeout == 0 {
+		timeout = defaultQMPTimeout
+	}
+	address, err := machineSocket(name, "qmp", "")
 	if err != nil {
 		return Monitor{}, err
 	}
+	monitor := Monitor{
+		Network: network,
+		Address: *address,
+		Timeout: timeout,
+	}
+	return monitor, nil
+}
+
+func machineSocket(name, prefix, suffix string) (*define.VMFile, error) {
+	rtDir, err := getRuntimeDir()
+	if err != nil {
+		return nil, err
+	}
 	if isRootful() {
 		rtDir = "/run"
 	}
 	rtDir = filepath.Join(rtDir, "podman")
 	if _, err := os.Stat(rtDir); errors.Is(err, fs.ErrNotExist) {
 		if err := os.MkdirAll(rtDir, 0755); err != nil {
-			return Monitor{}, err
+			return nil, err
 		}
 	}
-	if timeout == 0 {
-		timeout = defaultQMPTimeout
+	if prefix != "" {
+		name = prefix + "_" + name
 	}
-	address, err := define.NewMachineFile(filepath.Join(rtDir, "qmp_"+name+".sock"), nil)
-	if err != nil {
-		return Monitor{}, err
+	if suffix != "" {
+		name = name + "_" + suffix
 	}
-	monitor := Monitor{
-		Network: network,
-		Address: *address,
-		Timeout: timeout,
-	}
-	return monitor, nil
+	return define.NewMachineFile(filepath.Join(rtDir, name+".sock"), nil)
```

Comment:
fyi, there are actually plans for podman 5 to make a sister to `VMFile` but more like VMSocket or MachineSocket.

Reply:
I will refactor this PR if the changes lands before this one is merged.